File: Javascript/easy/Exercism/matching-brackets/matching-brackets.js
const BRACKET_MAP = {
    '{': '}',
    '[': ']',
    '(': ')'
};

export const isPaired = (str) => {
    if(str.length === 0) return true;
    let stack = [];

    for (let i = 0; i < str.length; i++) {

        if(str[i] in BRACKET_MAP) {
            stack.push(str[i]);
        } else if (Object.values(BRACKET_MAP).includes(str[i])){
            if(BRACKET_MAP[stack.pop()] !== str[i]) return false;
        }
    }

    return stack.length === 0;
};

File: Python/easy/Exercism/matching-brackets/matching_brackets.py
def is_paired(input_string):
    # Define matching pairs of brackets
    matching_bracket = {')': '(', ']': '[', '}': '{'}
    stack = []

    for char in input_string:
        if char in matching_bracket.values():  # If it's one of '(', '[', '{'
            stack.append(char)
        elif char in matching_bracket.keys():  # If it's one of ')', ']', '}'
            if stack and stack[-1] == matching_bracket[char]:
                stack.pop()
            else:
                return False

    return len(stack) == 0
